src/lib/reasoners/bitv.ml

```diff
@@ -784,36 +802,38 @@ module Shostak(X : ALIEN) = struct
             let c, eq, pat = apply_sub_rev bsub c [] [] req in
             eq, csub, if c then Some pat else None
         end
-      | st :: eq, n :: pt when st.sz < n ->
+      | { bv = S_Var { sorte = B; _ }; _ } as st :: eq,
+        n :: pt when st.sz < n ->
         (* Since we start with a [slicing_pattern], this should only occur when
            a B-variable has been split into parts below. *)
         slice_vars eq (n - st.sz :: pt) (true, st :: req, subs)
       | st :: eq, n :: pt when st.sz = n ->
         slice_vars eq pt (c, st :: req, subs)
       | st :: eq, n :: pt ->
-        let (st_n, rst, flag) = slice_var st n in
+        let (nvar_list, pat_tail, flag) = slice_var st n pt in
```

Comment:
Please keep the name `pt` instead of `pat_tail`. Yes `pt` may not be a good name, but it's better to have one name to refer to a thing rather than multiple names.

Reply:
Hm, I changed it because it is not the same one and I wanted to avoid shadowing it, `pt'` should be better?

Reply:
I think it would be an error to keep using the old `pt` here, and so shadowing is fine. Conceptually, `pt` refers to "the rest of the pattern" and after `slice_var` is called, it returns "the rest of the pattern". But if you insist on not shadowing, then yes, `pt'` is lower cognitive complexity because it's clearer that it is "the same kind of thing" as `pt'`.

Reply:
Interesting, I would say that `pt` is clearer to me because the fact that there is an assignment at all indicates that it is a modified version. Anyways, `pt'` it is here